inference-engine/src/mkldnn_plugin/mkldnn_node.cpp

```diff
@@ -636,6 +639,27 @@ void MKLDNNNode::execute(mkldnn::stream strm) {
     }
 }
 
+void MKLDNNNode::executeDynamic(mkldnn::stream strm) {
+    redefineOutputMemory();
+    executeDynamicBody(strm);
+}
+
+void MKLDNNNode::executeDynamicBody(mkldnn::stream strm) {
+    IE_THROW() << "[DS] executeDynamicBody not implemented for node with type: " << getTypeStr();
+}
+
+void MKLDNNNode::redefineOutputMemory(std::vector<std::vector<size_t>> newShapes) {
```

Comment:
What do you think, would it be better to have two such methods `redefineOutputMemory()` and `redefineOutputMemory(const std::vector<std::vector<size_t>>& newShapes)` that will implement these two separate branches? And name doesn't seems to be very descriptive m.b. something like `redefineOutputShape` or `resetOutputShape`?